src/main.zig

```diff
@@ -190,32 +205,37 @@ const Context = struct {
     entry: Entry,
     instance: Instance,
     device: Device,
+    physical_devices: []c.VkPhysicalDevice,
 
     fn init(allocator: std.mem.Allocator) !Self {
         var entry = try Entry.init();
         const instance = try Instance.init(entry, null);
         const physical_devices = try instance.get_physical_devices(allocator);
-        defer allocator.free(physical_devices);
         const queue_family = (try instance.select_queue_family(physical_devices, allocator, c.VK_QUEUE_COMPUTE_BIT)).?;
         const device = try Device.init(instance, queue_family, null);
         return .{
             .entry = entry,
             .instance = instance,
             .device = device,
+            .physical_devices = physical_devices,
         };
     }
 
-    fn deinit(self: *Self) void {
+    fn deinit(self: *Self, allocator: std.mem.Allocator) void {
         self.device.deinit();
         self.instance.deinit();
         self.entry.deinit();
+        allocator.free(self.physical_devices);
     }
 };
 
 pub fn main() !void {
     var general_purpose_allocator = std.heap.GeneralPurposeAllocator(.{}){};
+    defer std.debug.assert(.ok == general_purpose_allocator.deinit());
```

Comment:
Isn't it more natural to do it like this?
defer std.debug.assert(general_purpose_allocator.deinit() == .ok);

Reply:
I'm about to overwrite this commit and i've removed this line. I'll leave it to you to add it back now that its been brought to your attention.